File: app/src/main/java/com/example/gmailish/data/dao/LabelDao.java
package com.example.gmailish.data.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.gmailish.data.entity.LabelEntity;

import java.util.List;

@Dao
public interface LabelDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void upsert(LabelEntity label);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void upsertAll(List<LabelEntity> labels);

    // Existing: list by owner (generic)
    @Query("SELECT * FROM labels WHERE ownerId = :ownerId ORDER BY name ASC")
    List<LabelEntity> getByOwnerSync(String ownerId);

    // Matches InboxActivity.loadLocalLabels() usage exactly
    @Query("SELECT * FROM labels WHERE ownerId = :ownerId ORDER BY name ASC")
    List<LabelEntity> getAllByOwner(String ownerId);

    // Fetch by owner and name (used by LabelRepository.getLabelByName)
    @Query("SELECT * FROM labels WHERE ownerId = :ownerId AND name = :name LIMIT 1")
    LabelEntity getByName(String ownerId, String name);

    // Delete by id (used by LabelRepository.deleteLabel)
    @Query("DELETE FROM labels WHERE id = :labelId")
    int deleteById(String labelId);
}

File: app/src/main/java/com/example/gmailish/data/dao/LabelDao.kt
package com.example.gmailish.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.gmailish.data.entity.LabelEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface LabelDao {
    // Insert or replace one label
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun upsert(label: LabelEntity)

    // Insert or replace many labels
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun upsertAll(labels: List<LabelEntity>)

    // Observe all labels for an owner (reactive)
    @Query("SELECT * FROM labels WHERE ownerId = :ownerId ORDER BY name COLLATE NOCASE ASC")
    fun observeLabels(ownerId: String): Flow<List<LabelEntity>>

    // Get a label by name for an owner (one-shot)
    @Query("SELECT * FROM labels WHERE ownerId = :ownerId AND name = :name LIMIT 1")
    suspend fun getByName(ownerId: String, name: String): LabelEntity?

    // Delete a label by id
    @Query("DELETE FROM labels WHERE id = :labelId")
    suspend fun deleteById(labelId: String)
}